File: src/main/java/spring/springserver/domain/auth/interfaces/TokenProvider.java
package spring.springserver.domain.auth.interfaces;

import jakarta.servlet.http.HttpServletRequest;
import spring.springserver.domain.auth.data.request.GenerateTokenRequest;
import spring.springserver.domain.member.entity.Role;

public interface TokenProvider {

	String generateRefreshToken(GenerateTokenRequest generateTokenRequest);
	String generateAccessToken(GenerateTokenRequest generateTokenRequest);

	String getUsernameFromToken(String token);

	Role getRole(String token);

	String resolveToken(HttpServletRequest httpServletRequest);

	boolean isValidToken(String token);
}

File: src/main/java/spring/springserver/global/jwt/TokenProvider.kt
package spring.springserver.global.jwt

import jakarta.servlet.http.HttpServletRequest
import spring.springserver.domain.auth.data.request.GenerateTokenRequest
import spring.springserver.domain.member.entity.Role

interface TokenProvider {

    fun generateRefreshToken(generateTokenRequest: GenerateTokenRequest): String?

    fun generateAccessToken(generateTokenRequest: GenerateTokenRequest): String?

    fun getUsernameFromToken(token: String): String?

    fun getRole(token: String): Role?

    fun resolveToken(httpServletRequest: HttpServletRequest?): String?

    fun isNotValidToken(token: String): Boolean
}
